interface/src/app/tribe_client/genesets.js

```diff
@@ -1,67 +1,98 @@
-      $scope.limit = 10;
-    }],
-    replace: true,
-    restrict: 'E',
-    scope: {
-      organism: '='
-    },
-    templateUrl: 'tribe_client/geneset-search-form.tpl.html'
-  };
-}])
+      self.limit = 10;
+      self.sendToNetwork = function() {
+        self.selectedGenes = {};
+
+        var genes = self.selectedGeneset.geneObjs;
+        for (var i = 0; i < genes.length; i++) {
+          self.selectedGenes[genes[i].id] = genes[i];
+        }
+        CommonGeneFuncts.sendToNetwork(self, $state);
+      };
+    }
+  ]
+})
 
 
 // Search box for users to enter their search text into
 // and go search Tribe genesets.
-.directive('genesetSearchBar', ['Genesets', function(Genesets) {
-  return {
-    controller: ['$scope', function($scope) {
-      $scope.search = {};
-      $scope.search['organism__scientific_name'] = $scope.organism;
-      $scope.search['limit'] = $scope.limit;
+.component('genesetSearchBar', {
+  templateUrl: 'tribe_client/geneset-search-bar.tpl.html',
+  bindings: {
+    limit: '@',
+    organism: '@',
+    genesets: '='
+  },
+  controller: ['$log', 'Genesets', 'Gene', function($log, Genesets, Gene) {
+    var self = this;
+
+    self.search = {};
 
-      $scope.searchGenesets = function() {
-        Genesets.query($scope.search, function(data) {
-          $scope.genesetResultCount = data.meta.total_count;
-          $scope.genesets = data.objects;
+    self.searchGenesets = function() {
+      var qparams = {};
+      qparams['query'] = self.search.query;
+      qparams['organism__scientific_name'] = self.organism;
+      qparams['limit'] = self.limit;
+
+      Genesets.query(qparams, function(data) {
+        self.genesets = [];
+        var returnedGenesets = data.objects;
+        angular.forEach(returnedGenesets, function(geneset) {
+          var entrezIds = geneset.tip.genes.join(',');
+          Gene.get({'entrezid__in': entrezIds},
+            function success(response) {
+              geneset.geneObjs = response.objects;
+              self.genesets.push(geneset);
+            },
+            function error(err) {
+              $log.error('Failed to get gene information: ' + err);
+              self.errorMessage = 'Failed to get gene information from server';
+            }
+          );
         });
-      };
-    }],
-    replace: true,
-    restrict: 'E',
-    scope: {
-      limit: '=',
-      organism: '=',
-      genesets: '='
-    },
-    templateUrl: 'tribe_client/geneset-search-bar.tpl.html'
-  };
-}])
+      });
+    };
+  }]
+})
 
 // Directive for table containing search results
-.directive('genesetResultTable', [function() {
-  return {
-    replace: true,
-    restrict: 'E',
-    scope: {
-      // TODO: Implement pagination for geneset search results in
-      // case the number of results ($scope.genesetResultCount above)
-      // is greater than this 'limit' parameter (the maximum number of
-      // geneset results that the Tribe API should return at once).
```

Comment:
Ah, I see. You mean that probably users wouldn't care about search hits beyond `limit`? That's a good question, but I agree it's a topic for another day.